### PR TITLE
cpp: relocate the deprecated attribute to fix clang compilation (#299)

### PR DESCRIPTION
**Pull request by beku-epitome**
Please review the syntax, I'm no Jinja expert. See #299 for explanation.

### CODE REVIEW

File: src/nunavut/lang/cpp/templates/_composite_type.j2 [hunk 1]
```diff
@@ -28,10 +28,7 @@
 // +-------------------------------------------------------------------------------------------------------------------+
 {% endifuses -%}
 {{ composite_type.doc | block_comment('cpp-doxygen', 0, 120) }}
-{% if composite_type.deprecated -%}
-[[deprecated("{{ composite_type }} is reaching the end of its life; there may be a newer version available")]]
-{% endif -%}
-struct {{composite_type|short_reference_name}} final
+struct {{"[[deprecated(\"{} is reaching the end of its life; there may be a newer version available\")]]".format(composite_type) if composite_type.deprecated else ""}} {{composite_type|short_reference_name}} final
```
Comment (pavel-kirienko) [inline src/nunavut/lang/cpp/templates/_composite_type.j2:31]: How about this

```suggestion
struct {% if composite_type.deprecated -%}
[[deprecated("{{ composite_type }} is reaching the end of its life; there may be a newer version available")]]
{%- endif -%}
{{composite_type|short_reference_name}} final
```

Comment (beku-epitome) [reply src/nunavut/lang/cpp/templates/_composite_type.j2:31]: Fine for me, is the whitespace after the struct needed?

Comment (pavel-kirienko) [reply src/nunavut/lang/cpp/templates/_composite_type.j2]: Yes, it is quite essential.

File: src/nunavut/lang/cpp/templates/_composite_type.j2 [hunk 2]
```diff
@@ -28,10 +28,10 @@
 // +-------------------------------------------------------------------------------------------------------------------+
 {% endifuses -%}
 {{ composite_type.doc | block_comment('cpp-doxygen', 0, 120) }}
-{% if composite_type.deprecated -%}
+struct {% if composite_type.deprecated -%}
 [[deprecated("{{ composite_type }} is reaching the end of its life; there may be a newer version available")]]
 {% endif -%}
```
Comment (pavel-kirienko) [inline src/nunavut/lang/cpp/templates/_composite_type.j2:33]:
```suggestion
{%- endif -%}
```

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
Please bump the patch version also.

File: src/nunavut/lang/cpp/templates/_composite_type.j2 (hunk 2 above)
Comment (beku-epitome) [reply src/nunavut/lang/cpp/templates/_composite_type.j2]: Done

**Review by pavel-kirienko — APPROVED** [review]

**Review by thirtytwobits — APPROVED** [review]

**Comment by thirtytwobits** [comment]
Thanks for the contribution @beku-epitome ❤️

**Pull request merged**